src/vaadin-grid-sorter.js

```diff
@@ -170,19 +171,31 @@ class GridSorterElement extends ThemableMixin(DirMixin(PolymerElement)) {
   disconnectedCallback() {
     super.disconnectedCallback();
     this._isConnected = false;
+    this._wasDisconnected = true;
   }
 
   /** @private */
-  _pathOrDirectionChanged(path, direction, isConnected) {
-    if (path === undefined || direction === undefined || isConnected === undefined) {
+  _pathOrDirectionChanged(path, direction) {
+    if (path === undefined || direction === undefined) {
       return;
     }
 
-    if (isConnected) {
+    if (this.isConnected) {
       this.dispatchEvent(new CustomEvent('sorter-changed', { bubbles: true, composed: true }));
     }
   }
 
+  /** @private */
+  __isConnectedChanged(isConnected) {
+    if (isConnected) {
+      if (this._wasDisconnected) {
+        this._wasDisconnected = false;
+      } else {
+        this._pathOrDirectionChanged(this.path, this.direction);
```

Comment:
Calling another observer from `__isConnectedChanged` doesn't seem right. Probably it should be done within multi-property observer, i.e. `__isConnectedOrPathOrDirectionChanged(isConnected, path, direction)`

Reply:
`__isConnectedOrPathOrDirectionChanged(isConnected, path, direction)` doesn't seem right to me because `__isConnectedChanged(isConnected)` has extra logic. How about extracting a third method to call from both observers?

Reply:
Extracted method